### PR TITLE
Chore: Support external rules outside @sonarwhal

### LINKED ISSUE #713: Load external rules outside @sonarwhal
Right now to load an external rule, there is some requirement:
1. the rule is installed in the project and the name is `@sonarwhal/rule(s)-name-of-the-rule`
1. run directly sonar from the rule folder.
1. run sonar in a parent folder of the rule.

Also, to read those rules we are using a different method, because the external rules packages can contain multiple rules.

The Idea is to make `loadResource` smart enough to look for the external rules and detect if it is a package with multiple rules or not.

### PR DESCRIPTION
**Pull request by sarvaje**
Fix #713

## Pull request checklist

Make sure you:

- [x] Signed the [Contributor License Agreement](https://cla.js.foundation/sonarwhal/sonarwhal)
- [x] Followed the [commit message guidelines](https://sonarwhal.com/docs/contributor-guide/contributing/pull-requests/#commitmessages)

For non-trivial changes, please make sure you also:

- [ ] Added/Updated related documentation.
- [ ] Added/Updated related tests.

### CODE REVIEW

**Comment by alrra** [comment]
We need to start documenting things :P

**Comment by molant** [comment]
> We need to start documenting things :P

https://github.com/sonarwhal/sonarwhal/tree/master/docs 😛

The issue has more details on what needs to be changed. This is more of a `Chore` than a `Fix` IMO though.

**Comment by sarvaje** [comment]
I have changed the title and the commit :)

File: src/lib/utils/resource-loader.ts [hunk 1]
```diff
@@ -23,6 +23,10 @@ import { validate as validateRule } from '../config/config-rules';
 const debug: debug.IDebugger = d(__filename);
 const PROJECT_ROOT: string = findPackageRoot();
 const NODE_MODULES_ROOT: string = findNodeModulesRoot();
+const externalPackages: Array<string> = globby.sync(`{${process.cwd()},${NODE_MODULES_ROOT}/{.,@sonarwhal,sonarwhal-}}/{{sonarwhal,rule,connector,formatter}-*,.}/package.json`, { nodir: false });
```
Comment (alrra) [inline src/lib/utils/resource-loader.ts]:
```txt
 {.,@sonarwhal,sonarwhal-}}/{{sonarwhal,rule,connector,formatter}-*,.}/package.json`, { nodir: false }); 
                               ^ is not needed
```

File: src/lib/utils/resource-loader.ts [hunk 2]
```diff
@@ -109,6 +113,46 @@ export const tryToLoadFrom = (resourcePath: string): any => {
     return builder;
 };
 
+/**
+ * Check if it is a package with multiple resources.
+ */
+const hasMultipleResources = (resource, type: string) => {
+    switch (type) {
+        case TYPE.connector:
```
Comment (alrra) [inline src/lib/utils/resource-loader.ts]: @sarvaje Can you add more comments here explaining why these things are done. Thanks!

**Pull request closed without merging**